save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestService.kt

```diff
@@ -19,19 +22,28 @@ class TestService(private val configProperties: ConfigProperties) {
     @Autowired
     private lateinit var testRepository: TestRepository
 
+    @Autowired
+    private lateinit var testSuiteRepository: TestSuiteRepository
+
     /**
      * @param tests
      */
-    fun saveTests(tests: List<Test>) {
-        testRepository.saveAll(tests)
+    fun saveTests(tests: List<TestDto>) {
+        tests.map { testDto ->
+            testSuiteRepository.findById(testDto.testSuiteId).ifPresent {
+                testRepository.findByHash(testDto.hash).ifPresentOrElse(
+                    {},
+                    { testRepository.save(Test(testDto.hash, testDto.filePath, LocalDateTime.now(), it)) })
```

Comment:
Is `it` parameter of the outer lamda here? Please use named parameters, it's not really obvious